src/main/java/core/basesyntax/DataReaderImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataReaderImpl implements DataReader {
+    private final FileReader fileReader;

```

Comment:
The `FileReader` should not be a class-level dependency. Consider creating the `FileReader` instance within the `readFromFile` method to handle different file paths independently.